frontend/src/pages/AccountSend.vue

```diff
@@ -590,20 +943,110 @@ function useSendForm() {
       });
       void txNotify(tx.hash, ethersProvider);
       await tx.wait();
-      resetForm();
+      await resetForm();
     } finally {
       isSending.value = false;
       showAdvancedSendWarning.value = false;
     }
   }
 
-  function resetForm() {
+  // Send funds
+  async function onBatchSendFormSubmit() {
+    try {
+      // Form validation
+      for (let i = 0; i < Sends.value.length; i++) {
+        if (!Sends.value[i].receiver || !Sends.value[i].token || !Sends.value[i].amount)
+          throw new Error('please complete the form');
+      }
+      if (!signer.value) throw new Error('wallet not connected');
+      if (!umbra.value) throw new Error('Umbra instance not configured');
+
+      // Verify the recipient ID is valid. (This throws if public keys could not be found. This check is also
+      // done in the Umbra class `send` method, but we do it here to throw before the user pays for a token approval.
+      // This should usually be caught by the isValidId rule anyway, but is here again as a safety check)
+      const ethersProvider = provider.value as Provider;
+      for (let i = 0; i < Sends.value.length; i++) {
+        await umbraUtils.lookupRecipient(Sends.value[i].receiver as string, ethersProvider);
```

Comment:
remember that `await` in loops are very bad for performance, this should be refactored to a map with `Promise.all`. We should also consider a tracking issue to change these lookups to a single multicall for even better performance and less RPC usage

Reply:
yea, I have a terrible habit of this but should be fixed here https://github.com/ScopeLift/umbra-protocol/pull/491/commits/6bce5d4a1a5581d1757361ce405cd01d37814bdc

Reply:
you don't need the `async` keyword in your map, because we're synchronously returning a promise—the below snippet changes `async (send)` to just `(send)`
```typescript
await Promise.all(
  batchSends.value.map((send) => {
    return umbraUtils.lookupRecipient(send.receiver as string, ethersProvider);
  })
);
```

Reply:
Makes sense. Looks like I caught it in later commits though, latest code don't have the `async` https://github.com/ScopeLift/umbra-protocol/blob/96389cd21c05143a62fd4450470c0fc9e9b744e9/frontend/src/pages/AccountSend.vue#L989